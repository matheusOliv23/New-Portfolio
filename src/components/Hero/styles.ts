export const HeroStyles = `h-[90vh] transition-all duration-300 lg:flex-row flex-col-reverse flex items-center justify-around px-7 py-3 bg-no-repeat bg-fixed bg-cover bg-center`

export const ImgContainer = `flex flex-col items-center gap-5`

export const HeroImgStyles = `w-[15.5rem] md:w-[22rem] md:h-[22rem] flex items-center justify-center bg-cover bg-center pr-[0.5rem] bg-no-repeat ring-[5px] ring-inset ring-teal-500 rounded-[50%] h-[15.5rem]`
