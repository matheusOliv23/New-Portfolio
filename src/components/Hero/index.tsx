import Headline from "../Headline"
import SocialIcons from "../SocialIcons"
import { HeroImgStyles, HeroStyles, ImgContainer } from "./styles"

export default function Hero() {
  return (
    <section
      style={{
        backgroundImage: `linear-gradient(rgba(0,0,0,0.2), rgba(0,0,0,0.9)), url('/images/ales-nesetril-opac.png')`,
        backgroundBlendMode: "lighten"
      }}
      className={HeroStyles}
    >
      <Headline />
      <div className={ImgContainer}>
        <div
          style={{ backgroundImage: "url('/images/eu.webp')" }}
          className={HeroImgStyles}
        />

        <SocialIcons />
      </div>
    </section>
  )
}
