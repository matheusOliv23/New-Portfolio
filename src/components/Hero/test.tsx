import { render } from "@testing-library/react"

import Hero from "./index"

describe("<Hero />", () => {
  it("should  match snapshot", () => {
    const { container } = render(<Hero />)

    expect(container).toMatchSnapshot()
  })
})
