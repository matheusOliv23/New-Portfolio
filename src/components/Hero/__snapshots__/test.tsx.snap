// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should  match snapshot 1`] = `
<div>
  <section
    class="lg:h-[650px] h-auto m-auto flex flex-col lg:gap-32 gap-5 transition-all duration-300 px-7 py-3 bg-no-repeat bg-fixed bg-cover bg-center"
    style="background-blend-mode: lighten;"
  >
    <header
      class="flex w-full border-b-[0.3px] border-[#ffffff14] p-4 px-6 justify-end"
    >
      <nav>
        <ul
          class="flex items-center gap-9 justify-between"
        >
          <li
            class="hover:text-primary cursor-pointer"
          >
            Home
          </li>
          <li
            class="hover:text-primary cursor-pointer"
          >
            Sobre Mim
          </li>
          <li
            class="hover:text-primary cursor-pointer"
          >
            Projetos
          </li>
          <li
            class="hover:text-primary cursor-pointer"
          >
            Habilidades
          </li>
        </ul>
      </nav>
    </header>
    <div
      class="lg:flex-row flex-col-reverse flex items-center justify-around"
    >
      <div
        class="flex mt-4 flex-col lg:items-start z-50 gap-10 !opacity-100 items-center justify-center lg:text-start text-center"
      >
        <h1
          class="font-bold md:text-4xl text-2xl tracking-widest"
        >
          MATHEUS HENRIQUE OLIVEIRA
        </h1>
        <h2
          class="lg:text-4xl text-4xl text-[#8a8f98] font-medium tracking-widest mt-2"
        >
          Desenvolvedor 
          <strong
            class="text-primary"
          >
            Front-End
          </strong>
        </h2>
        <h3
          class="lg:text-2xl text-xl max-w-[32rem]"
        >
          Transformando sonhos em realidade por meio da tecnologia
        </h3>
        <div
          class="flex lg:mt-3 w-[13rem]  gap-6 justify-center items-center"
        >
          <button
            class="ease-in duration-200 tracking-[0.25rem] before:bg-[#27282c] uppercase ring-2 ring-inset hover:text-primary relative text-base font-bold duration-300  hover:shadow-2xl hover:text-white  py-4 px-7 rounded min-w-[3rem]  text-center flex items-center w-full justify-center bg-[#1e9bff] text-white undefined"
            type="button"
          >
            Quem sou eu
          </button>
        </div>
      </div>
      <div
        class="flex flex-col items-center gap-5"
      >
        <div
          class="w-[15.5rem] lg:w-[18rem] lg:h-[18rem] flex items-center justify-center bg-cover bg-center pr-[0.5rem] bg-no-repeat ring-[5px] ring-inset ring-teal-500 rounded-[50%] h-[15.5rem]"
          style="background-image: url(/images/eu.webp);"
        />
        <div
          class="flex gap-10 mb-2 mt-2"
        >
          <div
            class="flex hover:opacity-50 flex-col items-center justify-center"
          >
            <a
              href="https://www.instagram.com/matheus.olli23/"
              rel="noreferrer"
              target="_blank"
            >
              <span />
            </a>
            <span
              class="text-xs font-medium"
            >
              Instagram
            </span>
          </div>
          <div
            class="flex hover:opacity-50 flex-col items-center justify-center"
          >
            <a
              href="https://www.linkedin.com/in/matheushenrioliveira/"
              rel="noreferrer"
              target="_blank"
            >
              <span />
            </a>
            <span
              class="text-xs font-medium"
            >
              Linkedin
            </span>
          </div>
          <div
            class="flex hover:opacity-50 flex-col items-center justify-center"
          >
            <a
              href="https://github.com/matheusOliv23"
              rel="noreferrer"
              target="_blank"
            >
              <span />
            </a>
            <span
              class="text-xs font-medium"
            >
              Github
            </span>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
